src/blockchain_utilities/blockchain_stats.cpp

```diff
@@ -0,0 +1,326 @@
+  uint64_t prevtxs = 0, currtxs = 0;
+  uint64_t currblks = 0;
+  uint64_t totins = 0, totouts = 0, totrings = 0;
+  uint32_t minins = 10, maxins = 0;
+  uint32_t minouts = 10, maxouts = 0;
+  uint32_t minrings = 50, maxrings = 0;
+  uint32_t io, tottxs = 0;
+  uint32_t txhr[24] = {0};
+  unsigned int i;
+
+  for (uint64_t h = block_start; h < block_stop; ++h)
+  {
+    block_ser bs;
+    bs.block = db->get_block_from_height(h);
+    time_t tt = bs.block.timestamp;
+    char timebuf[64];
+    gmtime_r(&tt, &currtm);
+    if (currtm.tm_mday > prevtm.tm_mday || (currtm.tm_mday == 1 && prevtm.tm_mday > 27))
+    {
+      if (!prevtm.tm_year) {
+        prevtm = currtm;
+        goto skip;
+      }
+      if (prevtm.tm_mday == 1 && currtm.tm_mday > 27)
+          goto skip;
+      strftime(timebuf, sizeof(timebuf), "%Y-%m-%d", &prevtm);
+      prevtm = currtm;
+      std::cout << timebuf << "\t" << currblks << "\t" << h << "\t" << currtxs << "\t" << prevtxs + currtxs << "\t" << currsz << "\t" << prevsz + currsz;
+      prevsz += currsz;
+      currsz = 0;
+      currblks = 0;
+      prevtxs += currtxs;
+      currtxs = 0;
+      if (!tottxs)
+        tottxs = 1;
+      if (do_inputs) {
+        std::cout << "\t" << (maxins ? minins : 0) << "\t" << maxins << "\t" << totins / tottxs;
+        minins = 10; maxins = 0; totins = 0;
+      }
+      if (do_outputs) {
+        std::cout << "\t" << (maxouts ? minouts : 0) << "\t" << maxouts << "\t" << totouts / tottxs;
+        minouts = 10; maxouts = 0; totouts = 0;
+      }
+      if (do_ringsize) {
+        std::cout << "\t" << (maxrings ? minrings : 0) << "\t" << maxrings << "\t" << totrings / tottxs;
+        minrings = 50; maxrings = 0; totrings = 0;
+      }
+      tottxs = 0;
+      if (do_hours) {
+        for (i=0; i<24; i++) {
+          std::cout << "\t" << txhr[i];
+          txhr[i] = 0;
+        }
+      }
+      std::cout << ENDL;
+    }
+skip:
+    for (const auto& tx_id : bs.block.tx_hashes)
+    {
+      if (tx_id == crypto::null_hash)
+      {
+        throw std::runtime_error("Aborting: tx == null_hash");
+      }
+      transaction tx = db->get_tx(tx_id);
+      bs.txs.push_back(tx);
+      if (do_inputs) {
+        io = tx.vin.size();
+        if (io < minins)
+          minins = io;
+        else if (io > maxins)
+          maxins = io;
+        totins += io;
+      }
+      if (do_ringsize) {
+        const cryptonote::txin_to_key& tx_in_to_key
+	                   = boost::get<cryptonote::txin_to_key>(tx.vin[0]);
+		io = tx_in_to_key.key_offsets.size();
+        if (io < minrings)
+          minrings = io;
+        else if (io > maxrings)
+          maxrings = io;
+        totrings += io;
+      }
+      if (do_outputs) {
+        io = tx.vout.size();
+        if (io < minouts)
+          minouts = io;
+        else if (io > maxouts)
+          maxouts = io;
+        totouts += io;
+      }
+      tottxs++;
+    }
+    blobdata bd = t_serializable_object_to_blob(bs);
```

Comment:
FWIW you could get the blob and parse, it'd save a serialization. Not sure it's significant though since you've got disk I/O anyway.

Reply:
Ok, done. It saves significant CPU time; noticeable if the data was already cached anyway.
But the byte counts are smaller now too. I guess this is closer to reality though.